module/assessments/helper.js

```diff
@@ -508,15 +509,60 @@ module.exports = class AssessmentsHelper {
      * @method
      * @name create
      * @param {String} templateId - assessment template id. 
-     * @param {String} userId - Logged in user id.
+     * @param {String} userDetails - Logged in user id.
      * @param {Object} requestedData - request body data.
      * @returns {Object} - Create solution from assessment template. 
      */
 
-    static create( templateId,userId,requestedData ) {
+    static create( templateId,userDetails,requestedData ) {
         return new Promise(async (resolve, reject) => {
             try {
 
+              let solutionData = 
+              await solutionsHelper.solutionDocuments(
+                  {
+                      _id : templateId
+                  },[
+                      "subType",
+                      "entityType",
+                      "entityTypeId"
+                    ]
+              )
+
+              if( solutionData[0].subType === messageConstants.common.INDIVIDUAL ) {
+
+                    let entity = 
+                    await entitiesHelper.entityDocuments({
+                        "metaInformation.externalId" :  userDetails.userId
```

Comment:
@aman-tunerlabs - "metaInformation.externalId" is not indexed so it should not be a query key, would suggest having a user extension reference and keycloak userId in the document root in these cases and use that. 
Filter here should be user id in combination with the entity type.

Reply:
@aks30 Did not got this man .